src/pages/Trade/components/ChartCard/Chart.tsx

```diff
@@ -29,22 +41,132 @@ type Props = {
 	period: PeriodLabel;
 	synthPair: SynthPair;
 };
-type ChartProps = StateProps & Props;
 
-const Chart: FC<ChartProps> = ({ synthPair: { quote }, data, isLoading, period, synthsMap }) => {
+type ChartProps = StateProps & DispatchProps & Props;
+
+const Chart: FC<ChartProps> = ({
+	walletInfo,
+	fetchMyTradesRequest,
+	trades,
+	synthPair: { quote },
+	data,
+	isLoading,
+	period,
+	synthsMap,
+}) => {
 	const { colors } = useContext(ThemeContext);
 	const { t } = useTranslation();
 	const synthSign = synthsMap[quote.name] && synthsMap[quote.name].sign;
 
+	useEffect(() => {
+		fetchMyTradesRequest();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [walletInfo.currentWallet]);
+
+	useInterval(() => {
+		fetchMyTradesRequest();
+	}, DEFAULT_REQUEST_REFRESH_INTERVAL);
+
+	const ratesLength = (data?.rates ?? []).length;
+	const tradesLength = (trades ?? []).length;
+
+	const dataWithTrades = useMemo(() => {
+		if (trades.length > 0 && data.rates.length > 0 && walletInfo.currentWallet != null) {
```

Comment:
for trades.length and data.rates ^ you probably want to use `ratesLength/tradesLength`